IPtProxy.go/controller.go

```diff
@@ -93,36 +89,47 @@ type Controller struct {
 	SnowflakeSqsUrl       string
 	SnowflakeSqsCreds     string
 
+	stateDir  string
 	listeners map[string]*pt.SocksListener
 	shutdown  map[string]chan struct{}
 }
 
-func NewController(stateDir string) *Controller {
-	return &Controller{
-		LogLevel:      "ERROR",
-		StateDir:      stateDir,
-		EnableLogging: true,
+func NewController(stateDir string, enableLogging, unsafeLogging bool, logLevel string) *Controller {
+	c := &Controller{
+		stateDir: stateDir,
 	}
-}
 
-func (c *Controller) Init() {
-	if err := createStateDir(c.StateDir); err != nil {
-		log.Fatalf("Failed to set up state directory: %s", err)
+	if logLevel == "" {
+		logLevel = "ERROR"
 	}
-	if err := ptlog.Init(c.EnableLogging,
-		path.Join(c.StateDir, "ipt.log"), c.UnsafeLogging); err != nil {
-		log.Fatalf("Failed to set initialize log: %s", err.Error())
+
+	if err := createStateDir(c.stateDir); err != nil {
+		log.Printf("Failed to set up state directory: %s", err)
+		return nil
 	}
-	if c.LogLevel != "" {
-		if err := ptlog.SetLogLevel(c.LogLevel); err != nil {
-			log.Fatalf("Failed to set log level: %s", err.Error())
-		}
+	if err := ptlog.Init(enableLogging,
+		path.Join(c.stateDir, LogFileName), unsafeLogging); err != nil {
+		log.Printf("Failed to set initialize log: %s", err.Error())
+		return nil
+	}
+	if err := ptlog.SetLogLevel(logLevel); err != nil {
+		log.Printf("Failed to set log level: %s", err.Error())
+		ptlog.Warnf("Failed to set log level: %s", err.Error())
 	}
+
 	if err := transports.Init(); err != nil {
-		log.Fatalf("Failed to initialize transports: %s", err.Error())
+		ptlog.Warnf("Failed to initialize transports: %s", err.Error())
```

Comment:
Nice, this is also going to help with the problem of multiple initializations, in case the constructor is called twice for two different `Controller`s. Although maybe we should fix this upstream.